x/concentrated-liquidity/incentives_test.go

```diff
@@ -3535,6 +3527,81 @@ func (s *KeeperTestSuite) TestQueryAndClaimAllIncentives() {
 	}
 }
 
+func (s *KeeperTestSuite) TestClaimAllIncentivesForPosition() {
+	// Init suite for the test.
+	s.SetupTest()
+
+	var (
+		defaultAddress   = s.TestAccs[0]
+		secondAddress    = s.TestAccs[1]
+		defaultBlockTime = time.Unix(1, 1).UTC()
+	)
+
+	s.Ctx = s.Ctx.WithBlockTime(defaultBlockTime)
+	requiredBalances := sdk.NewCoins(sdk.NewCoin(ETH, sdk.NewInt(1_000_000)), sdk.NewCoin(USDC, sdk.NewInt(5_000_000_000)))
+	s.FundAcc(defaultAddress, requiredBalances)
+	s.FundAcc(secondAddress, requiredBalances)
+
+	// Create CL pool
+	pool := s.PrepareConcentratedPool()
+
+	// Set up position
+	positionIdOne, _, _, _, _, _, _, err := s.App.ConcentratedLiquidityKeeper.CreatePosition(s.Ctx, pool.GetId(), defaultAddress, requiredBalances, sdk.ZeroInt(), sdk.ZeroInt(), DefaultLowerTick, DefaultUpperTick)
+	s.Require().NoError(err)
+
+	// Set incentives for pool to ensure accumulators work correctly
+	testIncentiveRecord := types.IncentiveRecord{
+		PoolId:               pool.GetId(),
+		IncentiveDenom:       USDC,
+		IncentiveCreatorAddr: secondAddress.String(), // different address is going to create incentives
+		IncentiveRecordBody: types.IncentiveRecordBody{
+			RemainingAmount: sdk.NewDec(10_000_000),
+			EmissionRate:    sdk.NewDec(1), // 1 per second
+			StartTime:       s.Ctx.BlockTime(),
+		},
+		MinUptime: time.Nanosecond,
+	}
+	err = s.App.ConcentratedLiquidityKeeper.SetMultipleIncentiveRecords(s.Ctx, []types.IncentiveRecord{testIncentiveRecord})
+	s.Require().NoError(err)
+
+	// attempting to claim with same blocktime
+	collectedInc, forfeitInc, err := s.App.ConcentratedLiquidityKeeper.ClaimAllIncentivesForPosition(s.Ctx, positionIdOne)
+	s.Require().NoError(err)
+
+	// expected collectedIncentives in same blockTime = 0
+	s.Require().Equal(sdk.NewCoins(), collectedInc)
+	s.Require().Equal(0, len(forfeitInc))
+
+	// add 1min to current blocktime
+	s.Ctx = s.Ctx.WithBlockTime(s.Ctx.BlockTime().Add(time.Minute))
+
+	// attempting to claim after 1min has passed
+	collectedInc, forfeitInc, err = s.App.ConcentratedLiquidityKeeper.ClaimAllIncentivesForPosition(s.Ctx, positionIdOne)
+	s.Require().NoError(err)
+
+	// expected collectedIncentives after 1min = 59.999999999901820104usdc ~ 59usdc becasue 1usdc emitted every second
+	// foefeited incentives will be zero because 0.999999999901820104 truncated down is 0
```

Comment:
```suggestion
	// expected collectedIncentives after 1min = 59.999999999901820104usdc ~ 59usdc because 1usdc emitted every second
	// forfeited incentives will be zero because 0.999999999901820104 truncated down is 0
```